index.js

```diff
@@ -65,12 +65,21 @@ function fastifyRedis (fastify, options, next) {
   }
 
   if (!redisOptions.lazyConnect) {
-    return client.info((err, _) => {
-      return err ? next(err) : next()
-    })
-  } else {
-    next()
+    const onError = function (err) {
+      client.quit(() => next(err))
+    }
+
+    const onReady = function () {
+      client.off('error', onError)
+      next()
+    }
+
+    return client
+      .on('ready', onReady)
+      .on('error', onError)
   }
+
+  next()
```

Comment:
We want to call the `next()` without checking the connection only when the `lazyConnect` option has been provided right? 
In this case we are calling the `next()` here and another time when the client is in `ready`/`error` state.
Am i wrong?

Reply:
When specified `lazyConnect`, the if clause will stop at `return client` and this `next` is not reachable.